File: Blind150/3.Sliding Window/GO/alternation_groups_2.go
// https://leetcode.com/problems/alternating-groups-ii/description
func numberOfAlternatingGroups(colors []int, k int) int {

	N := len(colors)

	l := 0

	res := 0

	for r := 1; r < N+(k-1); r++ {

		if colors[r%N] == colors[(r-1)%N] {

			l = r
		}

		if r-l+1 == k {
			res++
			l++

		}

	}

	return res

}

File: Blind150/3.Sliding Window/Python/alternation_groups_2.py
# https://leetcode.com/problems/alternating-groups-ii/description

# CIRCULAR ARRAY

from typing import List

class Solution:
    def numberOfAlternatingGroups(self, colors: List[int], k: int) -> int:
        res = 0
        l = 0
        N = len(colors)

        for r in range(1, N+(k-1)):

            if colors[r%N] == colors[(r-1)%N]:
                l = r

            if r - l + 1 == k:

                res += 1
                l += 1
        return res
